c/src/core/engine.c

```diff
@@ -203,9 +204,9 @@ pn_transport_t *pn_connection_transport(pn_connection_t *connection)
 
 void pn_condition_init(pn_condition_t *condition)
 {
-  condition->name = pn_string(NULL);
-  condition->description = pn_string(NULL);
-  condition->info = pn_data(0);
+  condition->name = NULL;
+  condition->description = NULL;
+  condition->info = pni_data(0, 0);
```

Comment:
Why not lazily create `condition->info` the data is still somewhat large and info is hardly ever used.